### ISSUE TITLE
`istio-cni` plugin should implement CNI STATUS verb when most container runtimes impelment it.

### ISSUE THREAD

**Issue by bleggett** [istio/istio#53160]
**Describe the feature request**

Istio's CNI plugin (especially in/largely for ambient mode) should support the CNI STATUS verb, once the various container runtimes all support it.

- CNI STATUS spec: https://github.com/containernetworking/cni/blob/main/SPEC.md#status-check-plugin-status
- CRI-O already supports STATUS
- `containerd` is adding: https://github.com/containerd/go-cni/pull/114
-  Multus is TBD

Basically, this is just a boolean flag that indicates to the container runtime launching pods whether all the plugins in the chain are "ready" or not.

Implementing support for this will let our plugin indicate to the container runtime that it is not currently ready to process ADD requests, which should provide cleaner behavior in scenarios where e.g.  the istio node agent is not reachable from the istio plugin.

**Describe alternatives you've considered**

**Affected product area (please put an X in all that apply)**

[x] Ambient
[ ] Docs
[ ] Dual Stack
[ ] Installation
[x] Networking
[ ] Performance and Scalability
[ ] Extensions and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

**Affected features (please put an X in all that apply)**

[ ] Multi Cluster
[ ] Virtual Machine
[ ] Multi Control Plane

**Additional context**

**Comment by keithmattix** [comment]
Do we have an ETA for when containerd will have it?

**Comment by howardjohn** [comment]
Do we need the container runtime to support it for us to implement it? I would think we can just do it ~now, and then it just not used if the underlying platform doesn't implement it, which si fine. Once they pick up support, they get better visibility

**Comment by bleggett** [comment]
> Do we need the container runtime to support it for us to implement it? I would think we can just do it ~now, and then it just not used if the underlying platform doesn't implement it, which si fine. Once they pick up support, they get better visibility

Yeah in theory probably not. I would feel better about waiting until ~two of the most common runtime support it though.
